src/scripts/gcr-auth.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash 
+
+# Set sudo to work whether logged in as root user or non-root user
+if [[ $EUID == 0 ]]; then export SUDO=""; else export SUDO="sudo"; fi
+
+# configure Docker to use gcloud as a credential helper
+mkdir -p "$HOME/.docker"
+
+if [[ "$PARAM_REGISTRY_URL" == *"docker.pkg.dev" ]]; then
+    gcloud auth configure-docker --quiet --project "${!INDIRECT_PROJECT_ID}" "$PARAM_REGISTRY_URL"
```

Comment:
From my experience with the docker-orb migration, users tend to utilize variables even for parameters typed as strings (see list below). This will be a problem with `PARAM_REGISTRY_URL` here and potentially with other string parameters in this orb.

- https://github.com/CircleCI-Public/docker-orb/issues/151
- https://github.com/CircleCI-Public/docker-orb/issues/135
- https://github.com/CircleCI-Public/docker-orb/issues/136
- https://github.com/CircleCI-Public/docker-orb/issues/137

I go into details of why this is a problem post orb tools migration [here](https://github.com/CircleCI-Public/docker-orb/issues/135#issuecomment-1159137531). The CPE team discussed the problem at some point but never concluded, making it another excellent standardization topic.

My idea is to implement and import a generic utility function that expands all environment variables with a given prefix. I'm trying this approach in the `gcp-gke-orb`, and you can check the function [here](https://github.com/CircleCI-Public/gcp-gke-orb/blob/v1.0.0/src/scripts/utils.sh#L16) and how I'm using it [here](https://github.com/CircleCI-Public/gcp-gke-orb/blob/v1.0.0/src/scripts/create-cluster.sh#L5). 

At the moment, it's a script contained in the orb. This means we would have to copy and paste it into every orb where this functionality is needed. Should we agree on moving forward with the utility function, we could move it to the Orb Tools, for example, and import it from there.